docs/aws/audit/lambdamonitoring/rules/check_vpc_access_remediation.mdx

```diff
@@ -1,6 +1,251 @@
+           print(f"Function {function_name} is associated with a VPC.")
+       else:
+           print(f"Function {function_name} is not associated with a VPC.")
+   ```
+
+By following these steps, you can ensure that your AWS Lambda functions do not have access to VPC-only resources, thereby preventing potential misconfigurations.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the AWS Lambda console.
+
+2. In the navigation pane, choose "Functions". You will see a list of all your Lambda functions.
+
+3. Select the function you want to check. In the function's configuration page, scroll down to the "VPC" section. 
+
+4. If the function is not connected to a VPC, the "VPC" field will be set to "No VPC". If the function is connected to a VPC, you will see the details of the VPC, subnets, and security groups that the function has access to. If the function needs to access VPC-only resources but is not connected to a VPC, this is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all your Lambda functions. You can do this by using the AWS CLI command:
+
+```bash
+aws lambda list-functions --region your-region
+```
+Replace "your-region" with your actual AWS region.
+
+2. The output of this command will give you a list of all your Lambda functions. For each function, you need to check if it has VPC access. You can do this by describing the function configuration using the following command:
+
+```bash
+aws lambda get-function-configuration --function-name your-function-name --region your-region
+```
+Replace "your-function-name" with the actual name of your Lambda function and "your-region" with your actual AWS region.
+
+3. The output of this command will give you the configuration details of your Lambda function. Look for the "VpcConfig" section in the output. If the "VpcConfig" section is present and it has "SubnetIds" and "SecurityGroupIds", then your Lambda function has access to VPC resources.
+
+4. Repeat steps 2 and 3 for each Lambda function in your list. If any function does not have a "VpcConfig" section or if the "VpcConfig" section does not have "SubnetIds" and "SecurityGroupIds", then that function does not have access to VPC-only resources.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary AWS SDK for Python (Boto3) modules in your Python script. You will need the 'boto3' module to interact with AWS services and the 'json' module to parse the AWS service responses.
+
+```python
+import boto3
+import json
+```
+
+2. Create a session using your AWS credentials. Replace 'aws_access_key_id', 'aws_secret_access_key', and 'aws_session_token' with your actual AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>